gulpfile.js

```diff
@@ -10,37 +10,35 @@
 
 'use strict';
 
-const depcheck_lib = require('depcheck');
-const eslint_lib = require('gulp-eslint');
 const fs = require('fs-extra');
-const gulp = require('gulp');
 const mergeStream = require('merge-stream');
 const mocha = require('gulp-mocha');
 const path = require('path');
-const runSeq = require('run-sequence');
+const runSeqBase = require('run-sequence');
 const tslint_lib = require("gulp-tslint");
 const typescript = require('gulp-typescript');
 const typings = require('gulp-typings');
 
-function task(name, deps, impl) {
+function task(gulp, name, deps, impl) {
   if (gulp.hasTask(name)) {
     throw new Error(
         `A task with the name ${JSON.stringify(name)} already exists!`);
   }
   gulp.task(name, deps, impl);
 }
 
-module.exports.init = function() {
-  task('init', () => gulp.src("./typings.json").pipe(typings()));
+module.exports.init = function(gulp) {
+  task(gulp, 'init', () => gulp.src("./typings.json").pipe(typings()));
 }
 
-module.exports.depcheck = function depcheck(options) {
+module.exports.depcheck = function depcheck(gulp, options) {
+  const depcheck_lib = require('depcheck');
   const defaultOptions = {stickyDeps: new Set()};
   options = Object.assign({}, defaultOptions, options);
 
-  task('depcheck', () => {
+  task(gulp, 'depcheck', () => {
     return new Promise((resolve, reject) => {
-      depcheck_lib(__dirname, {ignoreDirs: []}, resolve);
+      depcheck_lib(process.cwd(), {ignoreDirs: []}, resolve);
```

Comment:
I don't think cwd is the right thing here... we really want to __dirname of the calling gulpfile